ui/uihandler.go

```diff
@@ -903,7 +899,14 @@ func Serve(baseurl, port, debugpath, uipath, staticRsrcDir string, datadir strin
 		fs := http.FileServer(http.Dir(dataDir))
 		http.Handle(uiPath+"data/", LogAndFilterDataRequest(http.StripPrefix(uiPath+"data", fs)))
 	}
-	fhttp.Serve(hostPort, debugpath)
+	addr := fhttp.Serve(hostPort, debugpath)
+	setHostAndPort(hostPort, addr)
+	uiMsg := fmt.Sprintf("UI starting - visit:\nhttp://%s%s", urlHostPort, uiPath)
+	if !strings.Contains(port, ":") {
```

Comment:
minor nit: not clear about the use case in which this will be invoked. maybe a comment would have been helpful?

Reply:
yes it's about putting "localhost" when no binding address is requested by the user